app/code/Magento/InventoryApi/Test/Api/SourceRepository/CreateTest.php

```diff
@@ -68,6 +68,54 @@ public function testCreate()
         AssertArrayContains::assert($expectedData, $this->getSourceDataById($sourceId));
     }
 
+    public function testCreateWithWrongCarrierLinks()
+    {
+        $expectedData = [
+            SourceInterface::NAME => 'source-name-1',
+            SourceInterface::CONTACT_NAME => 'source-contact-name',
+            SourceInterface::EMAIL => 'source-email',
+            SourceInterface::ENABLED => true,
+            SourceInterface::DESCRIPTION => 'source-description',
+            SourceInterface::LATITUDE => 11.123456,
+            SourceInterface::LONGITUDE => 12.123456,
+            SourceInterface::COUNTRY_ID => 'US',
+            SourceInterface::REGION_ID => 10,
+            SourceInterface::CITY => 'source-city',
+            SourceInterface::STREET => 'source-street',
+            SourceInterface::POSTCODE => 'source-postcode',
+            SourceInterface::PHONE => 'source-phone',
+            SourceInterface::FAX => 'source-fax',
+            SourceInterface::PRIORITY => 200,
+            SourceInterface::USE_DEFAULT_CARRIER_CONFIG => 0,
+            SourceInterface::CARRIER_LINKS => [
+                [
+                    SourceCarrierLinkInterface::CARRIER_CODE => 'no_exists_1',
+                    SourceCarrierLinkInterface::POSITION => 100,
+                ],
+                [
+                    SourceCarrierLinkInterface::CARRIER_CODE => 'no_exists_2',
+                    SourceCarrierLinkInterface::POSITION => 200,
+                ],
+            ],
+        ];
+        $serviceInfo = [
+            'rest' => [
+                'resourcePath' => self::RESOURCE_PATH,
+                'httpMethod' => Request::HTTP_METHOD_POST,
+            ],
+            'soap' => [
+                'service' => self::SERVICE_NAME,
+                'operation' => self::SERVICE_NAME . 'Save',
+            ],
+        ];
+
+        try {
+            $this->_webApiCall($serviceInfo, ['source' => $expectedData]);
+        } catch (\Exception $e) {
+            self::assertEquals(\Magento\Framework\Webapi\Exception::HTTP_BAD_REQUEST, $e->getCode());
+        }
+    }
+
```

Comment:
Also, need to fix the other api-functional builds
We do not run api-functional builds on Travis, so we need to run them on local machine
And now we have broken tests